maxim/tests/test_connection_retry_logic.py

```
@@ -0,0 +1,526 @@
+
+        http_error = HTTPError(response=mock_response)
+
+        with patch.object(
+            self.maxim_api.connection_pool, "get_session"
+        ) as mock_session_context:
+            mock_session = Mock()
+            mock_session_context.return_value.__enter__.return_value = mock_session
+            mock_session.request.side_effect = http_error
+
+            # HTTP errors should not be retried and should raise immediately
+            with self.assertRaises(Exception) as context:
```

Ruff (0.11.9)

Use `pytest.raises` instead of unittest-style `assertRaises`

Replace `assertRaises` with `pytest.raises`

(PT027)